code/expr_rewriter.go

```diff
@@ -101,13 +101,14 @@ func (r *Rewriter) rewriteInject(call *ast.CallExpr) (bool, ast.Stmt, error) {
 		argName = ast.NewIdent("_")
 	}
 
-	cond := ast.NewIdent("ok")
+	err := ast.NewIdent("err")
 	init := &ast.AssignStmt{
-		Lhs: []ast.Expr{argName, cond},
+		Lhs: []ast.Expr{argName, err},
 		Rhs: []ast.Expr{checkCall},
 		Tok: token.DEFINE,
 	}
 
+	cond := ast.NewIdent("err == nil")
```

Comment:
Feel a bit tricky because `err == nil` is not an `Ident`.

Reply:
I do not know much about the go AST, how do you suggest to improve this?

Reply:
```go
&ast.BinaryExpr{
    X:  err,
    Op: token.EQL,
    Y:  ast.NewIdent("nil"),
}
```